joomla/joomla-cms
# [4.0] Workflow stage delete publish

Pull Request for Issue #21846 

### Summary of Changes
Remove the method delete() from stageTable to stageModel.
Doing this some fixes have been added, comments and code have been improved

### Testing Instructions
Make a workflow with several stages and transitions.
Now try to change the status of one or more stages, thrash them, unpublish them.

If you try to trash a default state, you must get an alert.
Use the filter "trashed" and delete trashed stages.

Make sure that the transitions to and from the deletes stages are deleted too.


### Expected result
All changes are correct, all alerts are correct


### Actual result
The alert for trashing the default stage is wrong (see  #21846)
Trashing several stages results in wrong conditions for stages. 


### Documentation Changes Required
no
